canister/src/api/get_block_headers.rs

```diff
@@ -62,4 +233,313 @@ mod test {
+            .unwrap();
+
+        // The response should contain the header of `block`.
+        assert_eq!(
+            get_block_headers(GetBlockHeadersRequest {
+                start_height: 1,
+                end_height: Some(1),
+            })
+            .unwrap(),
+            GetBlockHeadersResponse {
+                tip_height: 1,
+                block_headers: vec![block_header_blob.clone()]
+            }
+        );
+
+        // The response should contain the header of `block`.
+        assert_eq!(
+            get_block_headers(GetBlockHeadersRequest {
+                start_height: 1,
+                end_height: None,
+            })
+            .unwrap(),
+            GetBlockHeadersResponse {
+                tip_height: 1,
+                block_headers: vec![block_header_blob.clone()]
+            }
+        );
+
+        // The response should contain headers of all blocks.
+        assert_eq!(
+            get_block_headers(GetBlockHeadersRequest {
+                start_height: 0,
+                end_height: Some(1),
+            })
+            .unwrap(),
+            GetBlockHeadersResponse {
+                tip_height: 1,
+                block_headers: vec![genesis_header_blob.clone(), block_header_blob.clone()]
+            }
+        );
+
+        // The response should contain headers of all blocks.
+        assert_eq!(
+            get_block_headers(GetBlockHeadersRequest {
+                start_height: 0,
+                end_height: None,
+            })
+            .unwrap(),
+            GetBlockHeadersResponse {
+                tip_height: 1,
+                block_headers: vec![genesis_header_blob.clone(), block_header_blob.clone()]
+            }
+        );
+    }
+
+    fn helper_initialize_and_get_heder_blobs(
+        stability_threshold: u128,
+        block_num: u32,
+        network: Network,
+    ) -> Vec<Vec<u8>> {
+        crate::init(Config {
+            stability_threshold,
+            network,
+            ..Default::default()
+        });
+        let genesis_block = genesis_block(network);
+
+        let mut prev_block_header = *genesis_block.header();
+        let mut genesis_header_blob = vec![];
+        genesis_block
+            .header()
+            .consensus_encode(&mut genesis_header_blob)
+            .unwrap();
+
+        let mut blobs = vec![genesis_header_blob];
+
+        // Genesis block is already added hence we need to add `block_num - 1`` more blocks.
+        for _ in 0..block_num - 1 {
+            let block = BlockBuilder::with_prev_header(&prev_block_header).build();
+            prev_block_header = *block.header();
+
+            let mut block_blob = vec![];
+            block.header().consensus_encode(&mut block_blob).unwrap();
+            blobs.push(block_blob);
+
+            with_state_mut(|state| insert_block(state, block).unwrap());
+        }
+
+        with_state_mut(ingest_stable_blocks_into_utxoset);
```

Comment:
Proptest is failing on this with: 
`Cannot insert outpoint OutPoint { txid: 4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b, vout: 0 }`
@ielashi did you see something similar, or do you know what can cause it?